src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java

```diff
@@ -315,26 +315,25 @@ else if (player.getTeam() == Team.BLUE) {
         }
     }
 
-
     // Used for test that units can get removed and then an endOf game state exists
     public GameDelta getCurrentGameDelta(){
         checkGameOver();
-        return new GameDelta(null, null, getGameOverInfo());
+        return new GameDelta(null, null, getGameOverInfo(false));
     }
 
     private Optional<Team> getWinnerTeam() {
         Optional<PlayerDecorator> playerWithUnits = getAllPlayersThat(player -> !player.getUnits().isEmpty()).findFirst();
         return playerWithUnits.map(BasePlayerDecorator::getTeam);
     }
 
-    private @Nullable GameOverInfo getGameOverInfo() {
+    private @Nullable GameOverInfo getGameOverInfo(boolean hasSurrendered) {
         if (running) {
             return null;
         }
         Optional<Team> winnerTeam = getWinnerTeam();
         if (winnerTeam.isPresent()) {
             List<Long> winners = getAllPlayersThat(player -> player.getTeam().equals(winnerTeam.get())).map(PlayerDecorator::getId).collect(Collectors.toList());
-            if (gameType == GameType.RANKED) {
+            if (gameType == GameType.RANKED && !hasSurrendered) {
```

Comment:
Does this mean that if someone surrenders in a ranked game scores do not update?

Reply:
> Does this mean that if someone surrenders in a ranked game scores do not update?

Well that would be wrong, else you can surrender, if you see that you will lose

Reply:
nope.
this means just that, when someone surrenders, the GameOverInfo does not hold this information for the client.
this information can be found in the SurrenderInfo then.

The GameOverInfo holds the information about a rankedScoreChange only when a game ended regularly.

Reply:
I see, makes sense!

Reply:
I seeee